test/e2e/config/vsphere-ci.yaml

```diff
@@ -179,6 +179,12 @@ providers:
 
 variables:
   KUBERNETES_VERSION: "v1.29.0"
+  KUBERNETES_VERSION_UPGRADE_FROM: "v1.28.0"
+  KUBERNETES_VERSION_UPGRADE_TO: "v1.29.0"
+  # ETCD_VERSION_UPGRADE_TO & COREDNS_VERSION_UPGRADE_TO are required but not used today
+  # by the upgrade test as the CAPV ClusterClass doesn't have corresponding variables.
+  ETCD_VERSION_UPGRADE_TO: "3.5.10-0"
+  COREDNS_VERSION_UPGRADE_TO: "v1.11.1"
```

Comment:
Change to v1.10.1
Open PR In CAPI to make these env vars optional
Also add upgrade/from to to vsphere-dev.yaml